src/main/groovy/wooga/gradle/wdk/unity/WdkUnityPlugin.groovy

```diff
@@ -114,6 +131,25 @@ class WdkUnityPlugin implements Plugin<Project> {
         extension.iosResourcePluginDir.convention(extension.pluginsDir.dir(WdkUnityPluginConventions.IOS_PLUGIN_DIRECTORY))
         extension.androidResourcePluginDir.convention(extension.pluginsDir.dir(WdkUnityPluginConventions.ANDROID_PLUGIN_DIRECTORY))
         extension.webGLResourcePluginDir.convention(extension.pluginsDir.dir(WdkUnityPluginConventions.WEBGL_PLUGIN_DIRECTORY))
+
+        extension.generateMetaFiles.convention(true)
+        // Try tp deduce the package directory within the Unity assets directories, where the root of the sources of the wdk are
+        // For example: Assets/Wooga/Foobar/*
+        extension.packageDirectory.convention(project.provider({
+            Directory result = null
+            if (extension.assetsDir.present) {
+                def woogaDir = extension.assetsDir.get()
+                def files = woogaDir.asFile.listFiles()
+                if (files != null &&  files.length == 1 && files[0].isDirectory()) {
+                    def packageDir = files[0]
+                    result = woogaDir.dir(packageDir.name)
+                }
+            }
+            if (result == null) {
+                logger.warn("Could not deduce the package directory for this wdk")
+            }
+            result
+        }))
```

Comment:
Why doing it this way around? You could also find all `package.json` files in the Assets directory and generate pack tasks for each of them. Similar to how we did it in paket. I think this method as is would create more harm. I would rather try to find the package.json (the first one as a start) or let this variable be configuable only. The whole system falls flat when a .meta file lands in the directory by the way.

Reply:
I will iterate on the test I wrote for this to check the other cases.